docs/decisions/0019-oep-58-atlas-translations-design.rst

```diff
@@ -0,0 +1,423 @@
+frontend static files on every XBlock load.
+
+The LTI Consumer XBlock embeds the translations in its ``web_fragment`` via
+the `LtiConsumerXBlock._get_statici18n_js_url`_ and
+`LtiConsumerXBlock.student_view`_ methods.
+
+In order to separate the XBlock translations from the platform, it's isolated
+in a separate ``gettext`` name space. For example, the Drag and Drop XBlock
+namespace is ``DragAndDropI18N``` which is hardcoded in multiple places such
+as:
+
+- `XBlock Makefile compile_translations rule`_
+- `XBlock compiled JavaScript text.js translations`_
+- `XBlock main JavaScript file`_
+
+This design has it's trade-offs but it's been widely adopted and out of the
+scope of this proposal. Therefore, we'll improve on the existing design to
+load the ``atlas`` JavaScript translations instead of the bundled ones.
+
+Decisions
+=========
+
+Proposed Design for edX Platform ``conf/locale`` translations
+-------------------------------------------------------------
+
+Update the ``make pull_translations`` command to use the ``atlas pull``
+if the ``OPENEDX_ATLAS_PULL`` environment variable is set.
+
+This has been the standard for all repositories as seen in both
+`course-discovery atlas integration`_ and
+`frontend-app-learning atlas integration`_.
+
+The updated `edx-platform pull_translations`_ would look like the following::
+
+  pull_translations:
+    git clean -fdX conf/locale
+    ifeq ($(OPENEDX_ATLAS_PULL),)
+      find conf/locale -type d -mindepth 1 -maxdepth 1 -exec rm -rf {} +  # Remove stale translation files
+      atlas pull $(ATLAS_PULL_ARGS) translations/edx-platform/conf/locale:conf/locale
+    else
+      i18n_tool transifex pull
+    endif
+    # The commands below are simplified for demonistration purposes
+    i18n_tool generate --verbose 1
+    i18n_tool validate --verbose
+    paver i18n_compilejs
+
+
+This mostly is a non-controversial change that has been done in other repos
+already.
+
+The next section is a little more intricate and requires more discussion.
+
+Proposed Design for XBlocks and Plugins
+---------------------------------------
+
+The proposed design for XBlocks and Plugins is to use the ``atlas pull``
+in a centrally managed way for all XBlocks and Plugins to circumvent the
+the need for managing the translations in each XBlock.
+
+The XBlock translations is already stored in the `openedx-translations repo`_
+and is accessible by the `openedx-atlas`_ command-line.
+
+
+New ``ENABLE_ATLAS_TRANSLATIONS`` Waffle Switch
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``ENABLE_ATLAS_TRANSLATIONS`` switch will be disabled by default.
+It will be used by the ``XBlockI18nService`` to determine which translations
+to use for XBlocks until the `OEP-58`_ is fully implemented and
+the non-atlas translations are removed.
+
+The non-XBlock plugins such as `edx-val`_ are out of the scope of this
+proposal and will be handled separately as stated in the :ref:`non-goals`
+section.
```

Comment:
I'm not sure exactly what you mean by this. By "allow fallbacking" do you mean allowing per-repo/package configuration of if we should use atlas or not? I foresee situations where we have *some* translations available via `openedx-translations`/`openedx-atlas` but they haven't reached parity with what is included in the repo for one reason or another, in which case `atlas` would still be able to pull something and wouldn't trigger an automatic fallback.

Reply:
The case which I'm concerned about is non-openedx XBlocks which I think would need fallback. For example, https://github.com/eduNEXT/xblock-limesurvey has a `locale` directory but it's not included in the https://github.com/openedx/openedx-translations repository.

The XBlockI18NService should be able to detect that there's no translations being pulled by `atlas` into the `plugins-locale` directory, therefore should use the `django.mo` from the XBlock itself.

In the case of `@openedx` XBlocks this shouldn't be a concern because all the translations will be synced into the new project on a regular basis.

Reply:
@brian-smith-tcril I've removed the Waffle Switch section because I think backward compatilbity should be there even after OEP-58 has been fully implemented to accommodate for third-party pull requests.